packages/react-components/react-tags/src/components/Tag/Tag.types.ts

```diff
@@ -1,17 +1,30 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
+import { Avatar } from '@fluentui/react-avatar';
 
 export type TagSlots = {
-  root: Slot<'div'>;
+  root: NonNullable<Slot<'div'>>;
+  content?: Slot<'span'>;
+  avatar?: Slot<typeof Avatar>;
+  icon?: Slot<'span'>;
+  primaryText?: Slot<'span'>;
+  secondaryText?: Slot<'span'>;
+  dismissButton?: NonNullable<Slot<'button'>>;
 };
 
 /**
  * Tag Props
  */
-export type TagProps = ComponentProps<TagSlots> & {};
+export type TagProps = ComponentProps<TagSlots> & {
+  size?: 'extra-small' | 'small' | 'medium';
+  shape?: 'rounded' | 'circular';
+  appearance?: 'filled-darker' | 'filled-lighter' | 'tint' | 'outline';
+  disabled?: boolean;
+  checked?: boolean;
+  dismissable?: boolean;
+};
 
 /**
  * State used in rendering Tag
  */
-export type TagState = ComponentState<TagSlots>;
-// TODO: Remove semicolon from previous line, uncomment next line, and provide union of props to pick from TagProps.
-// & Required<Pick<TagProps, 'propName'>>
+export type TagState = ComponentState<TagSlots> &
+  Required<Pick<TagProps, 'dismissable' | 'size' | 'shape' | 'disabled' | 'checked' | 'appearance'>>;
```

Comment:
nit: these should be in alphabetical order

```suggestion
  Required<Pick<TagProps, 'appearance' | 'checked' | 'disabled' | 'dismissable' | 'shape' | 'size'>>;
```